src/main/java/edu/hawaii/its/api/controller/OotbRestController.java

```diff
@@ -15,40 +18,50 @@
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
-import edu.hawaii.its.api.service.HttpRequestService;
 import edu.hawaii.its.api.service.OotbHttpRequestService;
 import edu.hawaii.its.groupings.access.UserContextService;
 import edu.hawaii.its.groupings.configuration.OotbStaticUserAuthenticationFilter;
+import edu.hawaii.its.groupings.configuration.Realm;
 import edu.hawaii.its.groupings.service.OotbActiveUserProfileService;
 import edu.hawaii.its.groupings.type.OotbActiveProfile;
 
 @RestController
+@Profile("ootb")
 @RequestMapping("/api/groupings/ootb")
 public class OotbRestController {
     private static final Log logger = LogFactory.getLog(OotbRestController.class);
-    private final PolicyFactory policy = Sanitizers.FORMATTING;;
-
-    @Value("${url.api.2.1.base}")
-    private String API_2_1_BASE;
-
+    private final PolicyFactory policy = Sanitizers.FORMATTING;
     private final UserContextService userContextService;
-
     private final OotbStaticUserAuthenticationFilter ootbAuthenticationFilter;
-
     private final OotbActiveUserProfileService ootbActiveUserProfileService;
-
     private final OotbHttpRequestService ootbHttpRequestService;
-
+    private final Realm realmService;
+    @Value("${url.api.2.1.base}")
+    private String API_2_1_BASE;
 
     // Constructor.
     public OotbRestController(UserContextService userContextService,
             OotbStaticUserAuthenticationFilter ootbAuthenticationFilter,
             OotbActiveUserProfileService ootbActiveUserProfileService,
-            OotbHttpRequestService ootbHttpRequestService) {
+            OotbHttpRequestService ootbHttpRequestService,
+            Realm realmService) {
         this.userContextService = userContextService;
         this.ootbAuthenticationFilter = ootbAuthenticationFilter;
         this.ootbActiveUserProfileService = ootbActiveUserProfileService;
         this.ootbHttpRequestService = ootbHttpRequestService;
+        this.realmService = realmService;
+    }
+
+    @PostConstruct()
+    public void init() {
+        if(realmService.isOotb()) {
+            String adminProfile = ootbActiveUserProfileService.findGivenNameForAdminRole();
+            if (adminProfile != null) {
+                updateActiveDefaultUser(adminProfile);
+            } else {
+                logger.error("Admin profile not found, unable to initiate default API request.");
+            }
+        }
```

Comment:
This seems to be something that should be called inside a configuration class rather than in a controller.

Reply:
I reorganize the configuration and dependency injections for now to make it call inside of a configuration class!